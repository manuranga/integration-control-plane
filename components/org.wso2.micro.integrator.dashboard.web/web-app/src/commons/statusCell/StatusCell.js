/*
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 *
 *
 */

import React from 'react';
import Badge from '@material-ui/core/Badge';
import {withStyles} from '@material-ui/core/styles';
import PropTypes from "prop-types";

const styles = theme => ({
    margin: {
        margin: theme.spacing(2)
    },
    greenBadge: {
        backgroundColor: "#00ff00",
        color: "white"
    },
    redBadge: {
        backgroundColor: "#ff0000",
        color: "white"
    }
});

function StatusCell(props) {
    const {classes} = props;
    const nodeData = props.data;
    var stateClass;
    if (nodeData === "Active") {
        stateClass = classes.greenBadge;
    } else {
        stateClass = classes.redBadge;
    }
    return <tr>
        <td><Badge classes={{badge: stateClass}}
                   className={classes.margin}
                   badgeContent={10} color="secondary" variant="dot"/>{nodeData}</td>
    </tr>;
}

export default withStyles(styles)(StatusCell);
StatusCell.propTypes = {
    classes: PropTypes.object.isRequired
};
