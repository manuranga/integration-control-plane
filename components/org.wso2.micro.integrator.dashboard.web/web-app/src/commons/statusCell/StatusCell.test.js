import React from 'react';
import renderer from 'react-test-renderer';
import StatusCell from "./StatusCell";

test('Test on green batch', () => {
    const component = renderer.create(
        <StatusCell data={"Active"}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});

test('Test on red batch', () => {
    const component = renderer.create(
        <StatusCell data={"Inactive"}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});
