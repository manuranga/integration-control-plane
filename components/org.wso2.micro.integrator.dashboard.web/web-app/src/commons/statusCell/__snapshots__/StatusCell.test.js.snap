// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on green batch 1`] = `
<tr>
  <td>
    <span
      className="MuiBadge-root StatusCell-margin-1"
    >
      <span
        className="MuiBadge-badge StatusCell-greenBadge-2 MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-dot"
      >
        
      </span>
    </span>
    Active
  </td>
</tr>
`;

exports[`Test on red batch 1`] = `
<tr>
  <td>
    <span
      className="MuiBadge-root StatusCell-margin-1"
    >
      <span
        className="MuiBadge-badge StatusCell-redBadge-3 MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-dot"
      >
        
      </span>
    </span>
    Inactive
  </td>
</tr>
`;
